app/controllers/discovered_hosts_controller.rb

```diff
@@ -50,7 +51,7 @@ def destroy
 
   def edit
     @host = ::ForemanDiscovery::HostConverter.to_managed(@host, true, false) unless @host.nil?
-    @host.attributes = @host.apply_inherited_attributes(params[:host]) unless params[:host].empty?
+    @host.attributes = @host.apply_inherited_attributes(discovered_host_params) unless discovered_host_params.empty?
```

Comment:
You may need to add a `:host` argument to discovered_host_params and pass that through as the second arg to `filter_params` to specify the top level hash, as it won't be the standard `:discovered_host`.  Note that the API controller _does_ use :discovered_host, so the arg should default to `nil` and not be set in that controller.